src/MomentoCacheClient.php

```diff
@@ -313,12 +314,25 @@ public function decrBy(string $key, int $value): Redis|int|false
         throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
     }
 
-    /**
-     * @throws Exception
-     */
+
     public function del(array|string $key, string ...$other_keys): Redis|int|false
     {
-        throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
+        if (is_array($key)) {
+            $keys = $key;
+        } else {
+            $keys = array_merge([$key], $other_keys);
+        }
+        $deletedKeys = 0;
+        foreach ($keys as $key) {
+            $getResult = $this->client->get($this->cacheName, $key);
+            if ($getResult->asHit()) {
+                $deleteResult = $this->client->delete($this->cacheName, $key);
+                if ($deleteResult->asSuccess()) {
+                    $deletedKeys++;
+                }
+            }
+        }
```

Comment:
This implementation is more correct since you verify which keys existed, but there's a race condition and it will be slower than just deleting the keys. In our other implementations we opted to just delete the keys directly without doing a `get` first. The function will return a possibly incorrect number of deleted items, but that's OK for v1.

Return the total number of keys passed in as opposed to total number of deleted keys. In the tests you can verify behavior for redis vs momento separately using a function to detect if momento tests are running.